src/components/lyne-toast/lyne-toast.tsx

```diff
@@ -0,0 +1,330 @@
+    return (
+      <button type='button' tabIndex={0} onClick={onClickFn} role={this._internalConfig.action.role} class={this._buttonClass()}>
+        {innerToastEl}
+      </button>
+    );
+  }
+
+  /**
+   * Creates the CSS classes for the action button.
+   * @returns A CssClassMap with two default CSS classes and all the CSS classes
+   * added by consumers in _internalConfig.action.cssClass.
+   */
+  private _buttonClass(): CssClassMap {
+    const fn: (acc: CssClassMap, curr: string) => CssClassMap = (acc: CssClassMap, curr: string) => {
+      acc[curr] = true;
+
+      return acc;
+    };
+
+    const classArray: string[] = getClassList(this._internalConfig.action.cssClass);
+
+    return {
+      'lyne-focusable': true,
+      'toast-button': true,
+      ...classArray.reduce(fn, {})
+    };
+  }
+
+  /**
+   * Triggers the action handler and dismiss the toast.
+   * @param action The action provided bu consumers in the toast config.
+   */
+  private _handleButtonClick(action: InterfaceToastAction): Promise<boolean> {
+    try {
+      action.handler();
+    } catch (e) {
+      console.error(e);
+    }
+
+    return this.dismiss(undefined, action.role);
+  }
+
+  /**
+   * Renders the action button (link/action/close icon).
+   */
+  private _renderAction(): JSX.Element {
+
+    switch (this._internalConfig.action.type) {
+      case 'link': {
+        return (
+          <a class='toast-link lyne-focusable' href={this._internalConfig.action.href} target='_blank' rel='noreferrer' tabIndex={0} role={this._internalConfig.action.role} onClick={this.dismiss.bind(this, null, 'link')}>
+            {this._internalConfig.action.label}
+          </a>
+        );
+
+      }
+      case 'action': {
+        const innerElement = <span class='toast-label'>{this._internalConfig.action.label}</span>;
+        const clickFn = this._handleButtonClick.bind(this, this._internalConfig.action);
+
+        return this._renderActionCommonButton(clickFn, innerElement);
+      }
+      case 'icon': {
+        const innerElement = <span class='toast-close' innerHTML={crossSmall}/>;
+        const clickFn = this.dismiss.bind(this, null, 'cancel');
+
+        return this._renderActionCommonButton(clickFn, innerElement);
+      }
+      default:
+        throw new Error('Invalid action!');
+    }
+  }
+
+  /**
+   * Handles ltr/rtl direction and sets the correct CSS class
+   */
+  private _getCSSClassFromPageDirection(): string {
+    const isElRTL = isRTL(this.el);
+
+    switch (this._internalConfig.horizontalPosition) {
+      case 'start': {
+        return isElRTL
+          ? 'right'
+          : 'left';
+      }
+      case 'end': {
+        return isElRTL
+          ? 'left'
+          : 'right';
+      }
+      default: {
+        return this._internalConfig.horizontalPosition;
+      }
+    }
+  }
+
+  public render(): JSX.Element {
+    const zIndex = this.overlayIndex + 60000;
```

Comment:
We probably have to document somehow why 60000 is chosen, or have a centralized file where the z-indexes for the overlays are stored?

Reply:
Added new file: src/global/z-index-list.ts